src/Symfony/Component/Form/Extension/HtmlSanitizer/Type/TextTypeHtmlSanitizerExtension.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\HtmlSanitizer\Type;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class TextTypeHtmlSanitizerExtension extends AbstractTypeExtension
+{
+    public function __construct(
+        private ContainerInterface $sanitizers,
+        private string $defaultSanitizer = 'default',
+    ) {
+    }
+
+    public static function getExtendedTypes(): iterable
+    {
+        return [TextType::class];
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setDefaults(['sanitize_html' => false, 'sanitizer' => null])
+            ->setAllowedTypes('sanitize_html', 'bool')
+            ->setAllowedTypes('sanitizer', ['string', 'null'])
+        ;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if (!$options['sanitize_html']) {
+            return;
+        }
+
+        $sanitizers = $this->sanitizers;
+        $sanitizer = $options['sanitizer'] ?? $this->defaultSanitizer;
```

Comment:
Is there a specific reason not to configure `$this->defaultSanitizer` as the default value in `configureOptions()`?

Reply:
Yes: to make it clear that `null` means "use the default value". I had removed the possibility to pass null before, and that make it impossible to say so.